framework/src/play/data/validation/ValidationPlugin.java

```diff
@@ -152,7 +162,7 @@ static void save(Request request, Response response) {
                 }
             }
             String errorsData = URLEncoder.encode(errors.toString(), "utf-8");
```

Comment:
I guess the order is wrong: you first need to encrypt and then to encode.

Reply:
refactored